src/clarity-angular/data/tree-view/tree-node.ts

```diff
@@ -1,142 +1,189 @@
+            state("expanded", style({
                 "height": "*",
                 "overflow-y": "hidden"
-            }))
-        ]),
-        transition("false => true", [
-            style({
-                "height": "*",
+            })),
+            state("collapsed", style({
+                "height": 0,
                 "overflow-y": "hidden"
-            }),
-            animate("0.2s ease-in-out")
+            })),
+            transition(
+                "expanded <=> collapsed", animate("0.2s ease-in-out")
+            )
         ])
-    ])],
-    host: {"[class.clr-tree--compact]": "isCompact"}
+    ]
+
 })
-export class TreeNode extends AbstractTreeSelection {
+export class TreeNode
+    extends AbstractTreeSelection
+    implements OnDestroy {
+
+    constructor(public nodeExpand: Expand,
+                @Optional() @SkipSelf() public parent: TreeNode,
+                public treeSelectionService: TreeSelectionService) {
+        super(parent);
+        if (this.parent) {
+            this.parent.register(this);
+        }
+    }
 
-    constructor(@Optional() @SkipSelf() parent: TreeNode,
-                @Optional() @SkipSelf() public treeSelectionService: TreeSelectionService) {
-        super(parent, treeSelectionService);
+    private _children: TreeNode[] = [];
+
+    get children(): TreeNode[] {
+        return this._children;
     }
 
-    @Input("clrTreeModel") model: any;
+    /* Registration */
 
-    @ContentChildren(TreeNode) _children: QueryList<TreeNode>;
+    checkIfChildNodeRegistered(node: TreeNode): boolean {
+        return (this.children.indexOf(node) > -1);
+    }
 
-    /**
-     * Generates the child TreeNodes array from the ContentChildren QueryList
-     * @returns {TreeNode[]|Array}
-     */
-    get children(): AbstractTreeSelection[] {
-        return this._children ? this._children.toArray().filter(child => this !== child) : [];
+    //TODO: This should ideally be in AbstractTreeSelection
+    //Tried doing this but ran into some issues and also ran out of time.
+    //Will get this done later.
+    register(node: TreeNode): void {
+        if (!this.checkIfChildNodeRegistered(node)) {
+            this.children.push(node);
+            if (this.selectable) {
+                if (this.selected) {
+                    node.parentChanged(this.selected);
+                }
+            }
+        }
     }
 
-    get nodeSelected(): boolean {
-        return this.selected;
+    //TODO: This should ideally be in AbstractTreeSelection
+    //Tried doing this but ran into some issues and also ran out of time.
+    //Will get this done later.
+    unregister(node: TreeNode): void {
+        const index = this.children.indexOf(node);
+        if (index > -1) {
+            this.children.splice(index, 1);
+        }
     }
 
-    @Input("clrTreeNodeSelected")
-    set nodeSelected(value: boolean) {
+    /* Selection */
+
+    private _selectionActivated: boolean = false;
```

Comment:
Why do you need this extra boolean? Isn't `this.treeSelectionService.selectable` enough?